axelrod/strategies/punisher.py

```diff
@@ -143,3 +143,42 @@ def strategy(self, opponent: Player) -> Action:
             return D
         else:
             return C
+
+
+class TrickyLevelPunisher(Player):
+    """
+    A player starts by cooperating however, after 10, 50 and 100 rounds
+    will defect if at any point the number of defections
+    by an opponent is greater than 20%, 10% and 5% respectively.
+
+    Names:
+
+    - Tricky Level Punisher: [Eckhart2015]_
+    """
+
+    name = 'Tricky Level Punisher'
+    classifier = {
+        'memory_depth': float('inf'), # Long Memory
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def strategy(self, opponent: Player) -> Action:
+        if len(opponent.history) < 10:
+            return C
+        if opponent.defections / (opponent.defections + opponent.cooperations) > 0.2:
+            return D
+        if len(opponent.history) < 50:
+            return C
+        if opponent.defections / (opponent.defections + opponent.cooperations) > 0.1:
+            return D
```

Comment:
There is no test that covers this possibility. Please add a test to ensure 100% coverage.